packages/devcmd-cli/src/index.ts

```diff
@@ -48,15 +48,41 @@ async function isDir(path: string): Promise<boolean> {
   }
 }
 
+async function startProcess(command: string, args: Array<string>, dirPath: string): Promise<number> {
+  const spawnOptions: SpawnOptions = {
+    stdio: 'inherit',
+    cwd: dirPath
+  };
+
+  const processPromise: Promise<number> = new Promise<number>((resolve, reject) => {
+    const processInstance: ChildProcess = spawn(command, args, spawnOptions);
+
+    processInstance.on("error", (err: Error): void => {
+      throw err;
+    });
+
+    processInstance.on("close", (code: number): void => {
+      if (code === 0) resolve(code);
+      else reject(code);
+    });
+  });
+
+  return processPromise;
+}
+
+function isError(err: any): err is Error {
+  return err instanceof Error;
+}
+
 // Note: if launching a node subprocess for the resolution should turn out to be a problem,
 //   we could also use the npm module "resolve" to find the path ourselves (and e.g. require it in-process).
 //   See https://yarnpkg.com/package/resolve
-async function runInDevCmdsDir(dirPath: string) {
+async function runInDevCmdsDir(dirPath: string): Promise<void> {
   const [, , ...args] = process.argv;
 
-  // TODO: use spawn or so instead
-  execFileSync("node", ["-e", `require('devcmd/from-cli').run(...process.argv.slice(1))`, ...args], {
-    cwd: dirPath,
-    stdio: "inherit",
-  });
-}
+  try {
+    await startProcess('node', ["-e", `require('devcmd/from-cli').run(...process.argv.slice(1))`, ...args], dirPath);
+  } catch (err) {
+    isError(err) ? abort(err.message) : process.exit(1);
```

Comment:
As @schierla pointed out, this there are different kinds of errors here.

If an actual error is caught (testable by calling `throw new Error("message")` in a devcmd, the error message + its stack trace gets printed. However if the promise rejects (testable by calling `process.exit(1)` within a devcmd) only the abort message `Process failed with exit code 1` gets printed.